pkg/components/pluggable/grpc.go

```diff
@@ -34,7 +35,30 @@ type GRPCClient interface {
 	Ping(ctx context.Context, in *proto.PingRequest, opts ...grpc.CallOption) (*proto.PingResponse, error)
 }
 
-type GRPCConnectionDialer = func(ctx context.Context, name string) (*grpc.ClientConn, error)
+// NewConverterFunc returns a function that maps from any error to a business error.
+// if the error is unknown it is kept as is, otherwise a converter function will be used.
+func NewConverterFunc(errorsConverters MethodErrorConverter) func(error) error {
+	return func(err error) error {
+		s, ok := status.FromError(err)
+		if !ok {
+			return err
+		}
+		convert, ok := errorsConverters[s.Code()]
+		if !ok {
+			return err
+		}
+		return convert(*s)
+	}
+}
+
+type GRPCConnectionDialer func(ctx context.Context, name string, opts ...grpc.DialOption) (*grpc.ClientConn, error)
```

Comment:
Are these changes part of the error handling PR or just a separate branch/merge artifact?

Reply:
I can remove that if necessary.

I was using it previously, because I tried to implement the error converter as a gRPC interceptor, but it ended up in being more complicated than it should be. 

Would you mind if I keep it? I think it still helpful

Reply:
It's fine with me; I just wanted to be sure that I hadn't missed something.